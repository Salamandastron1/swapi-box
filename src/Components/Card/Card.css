.card {
  border: 5px solid #FEDA4B;
  background-color: black;
  display: grid;
  border-radius: 20px;
  width: 225px;
  margin: 1em;
  height: 325px;
  color: #FEDA4B;
  justify-items: left;
  padding-left: 7px;
  margin-bottom: 6em;
  font-family: 'Russo One', sans-serif;
}

.favorited {
  color: #FEDA4B;
}

.mil-falc {
  background-color: black;
  height: 30px;
  display: block;
}

h3 {
  margin: 0;
}

button {
  background-color: black;
  border: none;
  color: white;
  display: inline-block;
}

.mil-falc:hover {
  height: 40px;
  transition: all .25s ease-in-out;
  cursor: pointer;
}
