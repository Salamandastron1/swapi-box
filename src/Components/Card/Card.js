import React from 'react';
import './Card.css';

const Card = (props) => {
  return(
    <div className='card'>
      <h3>{props.name}</h3>
      <div>
        <button>Favs</button>
      </div>
      <ul>
        <li>
          pop
        </li>
      </ul>
    </div>
  )
}

export default Card;