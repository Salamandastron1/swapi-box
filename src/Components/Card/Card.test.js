import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Card from './Card'

describe('Card', () => {
  let wrapper;
  const favoriteACard = jest.fn();
  beforeEach(() => {
    wrapper = shallow(<Card 
                name={'meow'}
                properties={['meow','meow','meow', 'meow']}
                favoriteACard={favoriteACard}
                favorited={true}
                />);
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should call favorite a card on click', () => {
    wrapper.find('button').simulate('click')

    expect(favoriteACard).toBeCalled()
  })
  it('should toggle class based on boolean', () => {
    expect(wrapper.find('.selected').length).toEqual(1)
  })
  it('should toggle class based on boolean', () => {
    wrapper = shallow(<Card 
                        name={'meow'}
                        properties={['meow','meow','meow', 'meow']}
                        favoriteACard={favoriteACard}
                        favorited={false}
                      />)
    expect(wrapper.find('.null').length).toEqual(1)
  })
});