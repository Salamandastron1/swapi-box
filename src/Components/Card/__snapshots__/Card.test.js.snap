// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches the snapshot 1`] = `
<div
  className="card"
>
  <h3 />
  <div>
    <button>
      Favs
    </button>
  </div>
  <ul>
    <li>
      pop
    </li>
  </ul>
</div>
`;
